cmd/nerdctl/rm.go

```diff
@@ -71,155 +49,7 @@ func rmAction(cmd *cobra.Command, args []string) error {
-	}
-
-	defer func() {
-		if errdefs.IsNotFound(retErr) {
-			retErr = nil
-		}
-		if retErr != nil {
-			return
-		}
-
-		if err := os.RemoveAll(stateDir); err != nil {
-			logrus.WithError(retErr).Warnf("failed to remove container state dir %s", stateDir)
-		}
-		if name != "" {
-			if err := namst.Release(name, id); err != nil {
-				logrus.WithError(retErr).Warnf("failed to release container name %s", name)
-			}
-		}
-		if err := hostsstore.DeallocHostsFile(dataStore, ns, id); err != nil {
-			logrus.WithError(retErr).Warnf("failed to remove hosts file for container %q", id)
-		}
-	}()
-	if anonVolumesJSON, ok := l[labels.AnonymousVolumes]; ok && removeAnonVolumes {
-		var anonVolumes []string
-		if err := json.Unmarshal([]byte(anonVolumesJSON), &anonVolumes); err != nil {
-			return err
-		}
-		volStore, err := getVolumeStore(globalOptions)
-		if err != nil {
-			return err
-		}
-		defer func() {
-			if _, err := volStore.Remove(anonVolumes); err != nil {
-				logrus.WithError(err).Warnf("failed to remove anonymous volumes %v", anonVolumes)
-			}
-		}()
-	}
-
-	task, err := container.Task(ctx, cio.Load)
-	if err != nil {
-		if errdefs.IsNotFound(err) {
-			if container.Delete(ctx, containerd.WithSnapshotCleanup) != nil {
-				return container.Delete(ctx)
-			}
-		}
-		return err
-	}
-
-	status, err := task.Status(ctx)
-	if err != nil {
-		if errdefs.IsNotFound(err) {
-			return nil
-		}
-		return err
-	}
-
-	switch status.Status {
-	case containerd.Created, containerd.Stopped:
-		if _, err := task.Delete(ctx); err != nil && !errdefs.IsNotFound(err) {
-			return fmt.Errorf("failed to delete task %v: %w", id, err)
-		}
-	case containerd.Paused:
-		if !force {
-			return statusError{fmt.Errorf("you cannot remove a %v container %v. Unpause the container before attempting removal or force remove", status.Status, id)}
-		}
-		_, err := task.Delete(ctx, containerd.WithProcessKill)
-		if err != nil && !errdefs.IsNotFound(err) {
-			return fmt.Errorf("failed to delete task %v: %w", id, err)
-		}
-	// default is the case, when status.Status = containerd.Running
-	default:
-		if !force {
-			return statusError{fmt.Errorf("you cannot remove a %v container %v. Stop the container before attempting removal or force remove", status.Status, id)}
-		}
-		if err := task.Kill(ctx, syscall.SIGKILL); err != nil {
-			logrus.WithError(err).Warnf("failed to send SIGKILL")
-		}
-		es, err := task.Wait(ctx)
-		if err == nil {
-			<-es
-		}
-		_, err = task.Delete(ctx, containerd.WithProcessKill)
-		if err != nil && !errdefs.IsNotFound(err) {
-			logrus.WithError(err).Warnf("failed to delete task %v", id)
-		}
-	}
-	var delOpts []containerd.DeleteOpts
-	if _, err := container.Image(ctx); err == nil {
-		delOpts = append(delOpts, containerd.WithSnapshotCleanup)
-	}
-
-	if err := container.Delete(ctx, delOpts...); err != nil {
-		return err
-	}
-	return err
+	return container.Remove(cmd.Context(), args, globalOptions, force, removeAnonVolumes, cmd.OutOrStdout())
```

Comment:
I didn't create a `RmCommandOptions` struct since it only has 2 boolean (force/volume) and to make other callers (e.g., prune, run) to easily reuse the `RemoveContainer` func (Instead of constructing a rmoptions).

I can change to add a `RmCommandOptions` if that's still preferred and change `Remove` to accept it (I think it's fine to keep the `RemoveContainer` signature).

Reply:
Can we make all commands consistently have the option struct?

Also, s/Rm/Remove/

Reply:
updated